lib/lists.nix

```
@@ -821,6 +821,47 @@ rec {
    */
   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
 
+  /* Check if list contains only unique elements. O(n^2) complexity.
+
+     Type: allUnique :: [a] -> bool
+
+     Example:
+       allUnique [ 3 2 3 4 ]
+       => false
+       allUnique [ 3 2 4 1 ]
+       => true
+   */
+  allUnique = list: (builtins.length (unique list) == builtins.length list);
```

```suggestion
  allUnique = list: (length (unique list) == length list);
```

In `lib` the preference is to import all symbols at the top (minor performance boost). `length` is already available so no need to import it.

Also make sure this is the case for the other functions.

---

done